src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.StaticAssets.targets

```diff
@@ -0,0 +1,172 @@
+         the paths to all the references packages and projects content roots.
+       * ResolveAspNetCoreStaticAssetsInputs: Collects all the static assets from different sources
+         * Current project.
+         * Referenced project.
+         * Referenced packages.
+  -->
+
+  <UsingTask
+    TaskName="Microsoft.AspNetCore.Razor.Tasks.GenerateAspNetCoreStaticAssetsManifest"
+    AssemblyFile="$(RazorSdkBuildTasksAssembly)"
+    Condition="'$(RazorSdkBuildTasksAssembly)' != ''" />
+
+  <PropertyGroup>
+    <GenerateAspNetCoreStaticAssetsManifestDependsOn>
+      ResolveAspNetCoreStaticAssetsInputs;
+      _ResolveAspNetCoreStaticAssetsConfiguration;
+      _CreateAspNetCoreStaticAssetInputsCacheFile
+    </GenerateAspNetCoreStaticAssetsManifestDependsOn>
+    <ResolveAspNetCoreStaticAssetsInputsDependsOn>_ResolveAspNetCoreStaticAssetsConfiguration</ResolveAspNetCoreStaticAssetsInputsDependsOn>
+  </PropertyGroup>
+
+  <Target Name="_ResolveAspNetCoreStaticAssetsConfiguration">
+  
+    <PropertyGroup>
+      <_GeneratedAspnetCoreStaticAssetInputsCacheFile>$(IntermediateOutputPath)Microsoft.AspNetCore.StaticAssets.cache</_GeneratedAspnetCoreStaticAssetInputsCacheFile>
+      <_GeneratedAspNetCoreStaticAssetsDevelopmentManifest>$(IntermediateOutputPath)Microsoft.AspNetCore.StaticAssets.xml</_GeneratedAspNetCoreStaticAssetsDevelopmentManifest>
+    </PropertyGroup>
+
+  </Target>
+
+  <Target 
+    Name="_CreateAspNetCoreStaticAssetInputsCacheFile"
+    DependsOnTargets="ResolveAspNetCoreStaticAssetsInputs;_ResolveAspNetCoreStaticAssetsConfiguration">
+    
+    <ItemGroup>
+      <!-- 
+      This is the list of inputs that will be used for generating the manifest used during development. 
+      -->
+      <_ExternalAspNetCoreStaticAssetManifestEntry
+        Include="%(AspNetCoreStaticAsset.Identity)"
+        Condition="'%(SourceType)' != ''">
+          <BasePath>%(AspNetCoreStaticAsset.BasePath)</BasePath>
+          <ContentRoot>%(AspNetCoreStaticAsset.ContentRoot)</ContentRoot>
+      </_ExternalAspNetCoreStaticAssetManifestEntry>
+    </ItemGroup>
+
+    <Hash ItemsToHash="@(_ExternalAspNetCoreStaticAssetManifestEntry)">
+      <Output TaskParameter="HashResult" PropertyName="_AspNetCoreStaticAssetCacheHash" />
+    </Hash>
+
+    <WriteLinesToFile 
+      Lines="$(_AspNetCoreStaticAssetCacheHash)" 
+      File="$(_GeneratedAspnetCoreStaticAssetInputsCacheFile)" 
+      Overwrite="True" 
+      WriteOnlyWhenDifferent="True" />
+
+    <ItemGroup>
+      <FileWrites Include="$(_GeneratedAspnetCoreStaticAssetInputsCacheFile)" />
+    </ItemGroup>
+
+  </Target>
+
+  <!-- 
+    This target generates a manifest for development time that includes information
+    about the base path for the referenced package and project static assets. The
+    manifest includes the content root and the base path for each of the referenced
+    packages and projects.
```

Comment:
Define `base path`? I don't know what it is :)

Reply:
Its the prefix of the url space under which all the files for a given content root get exposed. Longer explanation above.